enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java

```diff
@@ -337,6 +337,11 @@ public static JWTInfoDto generateJWTInfoDto(JSONObject subscribedAPI, JWTValidat
         String apiVersion = requestContext.getMatchedAPI().getAPIConfig().getVersion();
         jwtInfoDto.setApiContext(apiContext);
         jwtInfoDto.setVersion(apiVersion);
+        if (jwtInfoDto.getJwtValidationInfo().getClaims() != null
+                && jwtInfoDto.getJwtValidationInfo().getClaims().get("sub") != null) {
+            String sub = (String) jwtInfoDto.getJwtValidationInfo().getClaims().get("sub");
+            jwtInfoDto.setSub(sub);
+        }
         constructJWTContent(subscribedAPI, apiKeyValidationInfoDTO, jwtInfoDto);
```

Comment:
Don't we have the same value in apiKeyValidationInfoDTO's endusername?

Reply:
Getting a null for apiKeyValidationInfoDTO's endusername. We are setting the endusername in here https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L140
But this dto is getting reset in this line: https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L148

After this line, endusername resets to null

Reply:
Getting a null for apiKeyValidationInfoDTO's endusername. We are setting the endusername from this method: https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L140

But this dto get reset in this line: https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L148

After this line, endusername resets to null.